src/runtime/hexagon_cache_allocator.cpp

```diff
@@ -0,0 +1,165 @@
+#include "HalideRuntime.h"
+#include "mini_hexagon_dma.h"
+#include "HalideRuntimeHexagonDma.h"
+
+namespace Halide { namespace Runtime { namespace Internal {
+
+typedef struct hexagon_cache_block {
+    void *l2memory;
+    size_t bytes;
+    bool used;
+    struct hexagon_cache_block *next;
+} hexagon_cache_pool_t;
+
+typedef hexagon_cache_pool_t *pcache_pool;
+
+pcache_pool hexagon_cache_pool = NULL;
+halide_mutex hexagon_cache_mutex; 
+
+}}}
+
+using namespace Halide::Runtime::Internal;
+
+extern "C" {
```

Comment:
Move this down below the internal linkage functions. These internal linkage functions should also be in a namespace Halide::Runtime::Internal::Hexagon

Reply:
Agreed